app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -0,0 +1,54 @@
+package ar.com.wolox.android.example.ui.login;
+
+import android.support.annotation.NonNull;
+
+import javax.inject.Inject;
+
+import ar.com.wolox.android.example.utils.Constants;
+import ar.com.wolox.android.example.utils.StringUtils;
+import ar.com.wolox.wolmo.core.presenter.BasePresenter;
+import ar.com.wolox.wolmo.core.util.SharedPreferencesManager;
+
+/**
+ * The Presenter(P) in the MVP architectural pattern for Login featureclwea
+ */
+public class LoginPresenter extends BasePresenter<ILoginView> {
+
+    private SharedPreferencesManager mSharedPreferencesManager;
+
+    @Inject
+    public LoginPresenter(SharedPreferencesManager mSharedPreferencesManager) {
+        this.mSharedPreferencesManager = mSharedPreferencesManager;
+    }
+
+    /**
+     * Logs in an user;
+     *
+     * @param email    Input email
+     * @param password Input password
+     */
+    void login(@NonNull String email, @NonNull String password) {
```

Comment:
I would prefer a structure like If -> else If. What's your opinion @DemianIac ?

Reply:
With the if -- else structure you can check the email first and then the password , showing one error at a time